src/Classifier.php

```diff
@@ -48,44 +283,109 @@ class Classifier
-
-        switch (true) {
-            case (in_array(strtolower($parseResult->device->family), $this->tablets)):
-                $classifyResult->matchType      = 'device';
-                $classifyResult->isTablet       = true;
-                $classifyResult->isMobileDevice = true;
-                $classifyResult->isComputer     = false;
-                break;
-
-            case (in_array(strtolower($parseResult->device->family), $this->mobileDevices)):
-                $classifyResult->matchType      = 'device';
-                $classifyResult->isMobileDevice = true;
-                $classifyResult->isMobile       = true;
-                $classifyResult->isComputer     = false;
-                break;
-
-            case (strtolower($parseResult->device->family) == 'spider'):
-                $classifyResult->matchType  = 'device';
-                $classifyResult->isSpider   = true;
-                $classifyResult->isComputer = false;
-                break;
-
-            case (in_array(strtolower($parseResult->os->family), $this->mobileOSs)):
-                $classifyResult->matchType      = 'os';
-                $classifyResult->isMobileDevice = true;
-                $classifyResult->isMobile       = true;
-                $classifyResult->isComputer     = false;
-                break;
-
-            case (in_array(strtolower($parseResult->ua->family), $this->mobileBrowsers)):
-                $classifyResult->matchType      = 'ua';
-                $classifyResult->isMobileDevice = true;
-                $classifyResult->isMobile       = true;
-                $classifyResult->isComputer     = false;
-                break;
+        $result = new Result();
+
+        foreach ($this->getRules() as $rule) {
+            if (isset($rule['device']) &&
+                preg_match('/^' . $rule['device'] . '$/i', $parseResult->device->family) === 0
+            ) {
+                continue;
+            }
+            if (isset($rule['os']) &&
+                preg_match('/^' . $rule['os'] . '$/i', $parseResult->os->family) === 0
+            ) {
+                continue;
+            }
+            if (isset($rule['ua']) &&
+                preg_match('/^' . $rule['ua'] . '$/i', $parseResult->ua->family) === 0
+            ) {
+                continue;
+            }
+
+            $result->isMobileDevice = in_array($rule['class'], ['tablet', 'mobile']);
+            $result->isMobile       = $rule['class'] === 'mobile';
+            $result->isTablet       = $rule['class'] === 'tablet';
+            $result->isSpider       = $rule['class'] === 'spider';
+            $result->isComputer     = $rule['class'] === 'desktop';
+            $result->matchType      = $rule['class'];
+
+            break;
+        }
+
+        return $result;
+    }
+
+    private function getRules() {
+        if ($this->rules) {
+            return $this->rules;
+        }
+
+        // Spider rules
+        array_push($this->rules, [
```

Comment:
This seems a long-winded way of writing `$this->rules[] = ...`

And according to the PHP Manual, it has an unnecessary overhead too.